### ISSUE TITLE
Question about contrastiveloss 

### ISSUE THREAD

**Issue by zjuxiaobaiq** [kinux98/SSL_ELN#4]
hi，Congratulations on your outstanding work. I wonder if this is reasonable  that when computing contrastiveloss the query features come from teacher model while key features come from students model? 
code : pxl_dist_aug = pixelwisecontrastiveloss(self, feature_vec_un_k.detach(), feature_vec_un_aug,  final_candid, final_indices)

**Comment by kinux98** [comment]
The pixel-wise feature embedding of the teacher network is provided as a supervision of the pixel-wise feature embedding of the student network. Just as pseudo labels for cross entropy computation (supervision for student network learning) are derived from teacher network prediction, pseudo pixel-wise feature embedding labels for contrastive loss computation are derived from teacher network's pixel-wise feature embedding.
